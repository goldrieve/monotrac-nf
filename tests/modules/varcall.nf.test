nextflow_process {

    name "Test Process VARCALL"
    script "main.nf"
    process "VAR_CALL"

    test("Should run variant calling") {

        when {
            params {
                reference = "${projectDir}/data/references/padded_targets_positive.fasta"
                model = "${projectDir}/data/model/r1041_e82_400bps_hac_v520"
                depth = 20
		        mpq = 5
		        blank = "$projectDir/data/references/blank.vcf.gz"

            }
            process {
                """
                input[0] = ["SVP_test", file("${projectDir}/data/syn_reads/SVP.fq.gz")] 
                input[1] = params.reference
                input[2] = params.model
                input[3] = params.depth
                input[4] = params.mpq
                input[5] = params.blank
                """
            }
        }

        then {
            assert process.success
            assert path(process.out[0][0][1]).linesGzip.contains('Tb927.1.3230\t2858\t.\tA\tG\t36.68\tPASS\tF\tGT:GQ:DP:AD:AF\t1/1:36:23:0,20:0.8696')
            assert path(process.out[0][0][1]).linesGzip.contains('Tb927.5.1220\t2934\t.\tC\tT\t43.94\tPASS\tF\tGT:GQ:DP:AD:AF\t1/1:43:23:0,21:0.9130')
        }

    }

}
