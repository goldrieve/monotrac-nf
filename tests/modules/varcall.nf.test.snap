{
    "Should run variant calling": {
        "content": [
            [
                [
                    "SVP_test",
                    "SVP_test_final.vcf.gz:md5,024f6d6a7c25e4e8aa6f850bd5e924c0",
                    "SVP_test_final.vcf.gz.csi:md5,0b171c8ff6dd7506c0cdd8f2bf790701"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T15:27:38.365335287"
    }
}